runtime/src/blueoil_image.cpp

```diff
@@ -0,0 +1,194 @@
+	    }
+	}
+    }
+    return dstTensor;
+}
+
+/*
+ * Resize Image
+ */
+Tensor ResizeHorizontal(Tensor &tensor, const int width,
+			const enum ResizeFilter filter) {
+    if (filter != RESIZE_FILTER_NEAREST_NEIGHBOR) {
+	throw "unknown ResizeFilter:"; // XXX
+    }
+    const int srcHeight = tensor.shape[0];
+    const int srcWidth  = tensor.shape[1];
+    const int channels  = tensor.shape[2];
+    const int height = srcHeight;
+    Tensor dstTensor = Tensor::zeros({height, width, channels});
+    float xScale = (float) width / (float) srcWidth;
+    int xSrcWindow = floorf(1/xScale);
+    xSrcWindow = 1; // Nearest
+    //xSrcWindow = (xSrcWindow < 2)? 2 :xSrcWindow; // Bi-Liner
+    for (int dstY = 0 ; dstY < height ; dstY++) {
+	for (int dstX = 0 ; dstX < width ; dstX++) {
+	    int srcX = (int) floorf(dstX/xScale);
+	    int srcY = dstY;
+	    for (int c = 0 ; c < channels ; c++) {
+		float v = 0.0;
+		float totalW = 0.0;
+		for (int x = -xSrcWindow ; x < xSrcWindow; x++){
+		    float *srcRGB = blueoil::image::Tensor_at(tensor,
+							      srcX + x, srcY);
+		    float d = fabs((float)x / (float)xSrcWindow);
+		    float w = (d<0.5)?1.0:0.0; // NearestNeighbor
+		    // float w = 1.0 - d; // Bi-Linear
+		    v += w * srcRGB[c];
+		    totalW += w;
+		}
+		float *dstRGB = blueoil::image::Tensor_at(dstTensor,
+							  dstX, dstY);
+		dstRGB[c] = v / totalW;
+	    }
+	}
+    }
+    return dstTensor;
+}
+
+Tensor ResizeVertical(Tensor &tensor, const int height,
+		      const enum ResizeFilter filter) {
+    if (filter != RESIZE_FILTER_NEAREST_NEIGHBOR) {
+	throw "unknown ResizeFilter:"; // XXX
+    }    
+    const int srcHeight = tensor.shape[0];
+    const int srcWidth  = tensor.shape[1];
+    const int channels  = tensor.shape[2];
+    const int width = srcWidth;
+    Tensor dstTensor = Tensor::zeros({height, width, channels});
+    float yScale = (float) height / (float) srcHeight;
+    int ySrcWindow = floorf(1/yScale);
+    ySrcWindow = 1; // Nearest
+    // ySrcWindow = (ySrcWindow < 2)? 2 :ySrcWindow; // Bi-Linear
+    for (int dstY = 0 ; dstY < height ; dstY++) {
+	for (int dstX = 0 ; dstX < width ; dstX++) {
+	    int srcX = dstX;
+	    int srcY = (int) floorf(dstY/yScale);
+	    for (int c = 0 ; c < channels ; c++) {
+		float v = 0.0;
+		float totalW = 0.0;
+		for (int y = -ySrcWindow ; y < ySrcWindow ; y++) {
+		    float *srcRGB = blueoil::image::Tensor_at(tensor,
+							      srcX, srcY + y);
+		    float d = fabs((float)y / (float)ySrcWindow);
+		    float w = (d<0.5)?1.0:0.0; // NearestNeighbor
+		    // float w = 1.0 - d; // Bi-Linear
+		    v += w * srcRGB[c];
+		    totalW += w;
+		}
+		float *dstRGB = blueoil::image::Tensor_at(dstTensor,
+							  dstX, dstY);
+		dstRGB[c] = v / totalW;
+	    }
+	}
+    }
+    return dstTensor;
+} 
+
+Tensor Resize(const Tensor& image, const int width, const int height,
+	      const enum ResizeFilter filter) {
+    // int height = image.shape[0];
+    // int width  = image.shape[1];
```

Comment:
If we do not need it let's remove it, if we need it why it is commented?

Reply:
I removed Its.
https://github.com/blue-oil/blueoil/pull/9/commits/3c9215681f325ce680e20949907ab409337d9d1c#diff-37230609db7212ca72dd52076761c190L172